framework/log/Dispatcher.php

```diff
@@ -184,24 +184,41 @@ public function dispatch($messages, $final)
     {
         $targetErrors = [];
         foreach ($this->targets as $target) {
-            if ($target->enabled) {
-                try {
-                    $target->collect($messages, $final);
-                } catch (\Exception $e) {
-                    $target->enabled = false;
-                    $targetErrors[] = [
-                        'Unable to send log via ' . get_class($target) . ': ' . ErrorHandler::convertExceptionToVerboseString($e),
-                        Logger::LEVEL_WARNING,
-                        __METHOD__,
-                        microtime(true),
-                        [],
-                    ];
-                }
+            if (!$target->enabled) {
+                continue;
+            }
+            try {
+                $target->collect($messages, $final);
+            } catch (\Throwable $t) {
+                $target->enabled = false;
+                $targetErrors[] = $this->generateTargetFailErrorMessage($target, $t, __METHOD__);
+            } catch (\Exception $e) {
+                $target->enabled = false;
+                $targetErrors[] = $this->generateTargetFailErrorMessage($target, $e, __METHOD__);
             }
         }
 
         if (!empty($targetErrors)) {
             $this->dispatch($targetErrors, true);
         }
     }
+
+    /**
+     * Generate target error message
+     *
+     * @param Target                $target
+     * @param \Throwable|\Exception $throwable
+     * @param string                $method
+     * @return array
```

Comment:
Need `@since` tag. Also, alignment should not be done for phpdoc.

Reply:
modified.